### PR TITLE
New server failure framework

### PR DESCRIPTION
**Pull request by einarmo**
Migrates (mostly) the old way to do failures to this new system, which lets us do random and timed errors in a consistent way.

### CODE REVIEW

File: Server/Program.cs [hunk 1]
```diff
@@ -101,9 +101,10 @@ internal sealed class ServerOptions
         [CommandLineOption("Set server redundancy support. One of None, Cold, Warm, Hot, Transparent, HotAndMirrored")]
         public string RedundancySupport { get; set; }
 
-        [CommandLineOption("Server issue: This is the denominator for a probability that an arbitrary browse operation will fail " +
-            "I.e. 5 means that 1/5 browse ops will fail with BadNoCommunication")]
-        public int RandomBrowseFail { get; set; }
+        [CommandLineOption("Server issue: This is the denominator for a probability that an arbitrary operation will fail " +
+            "I.e. 5 means that 1/5 browse ops will fail with BadInternalError")]
+        public int RandomFail { get; set; }
```
[the remark this answers is not in the document]
Comment (einarmo) [reply Server/Program.cs:106]: The server isn't really part of the product.

[the remark this answers is not in the document]
Comment (einarmo) [reply Server/Program.cs:106]: I don't want to make guarantees of compatibility here.

[the remark this answers is not in the document]
Comment (ozangoktan) [reply Server/Program.cs:106]: Sure, that's fair.

**Review by ozangoktan — APPROVED** [review]
LGTM

**Pull request merged**